### PR TITLE
Pipeguns Round 2; combining jamming/misfire, bug fixing also more gold paint

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Consolidates the misfire and jamming behaviours into the same system, wherein, depending on which variables are set to true, you can set a variety of different behaviours including just jamming (mosins), just misfiring (det revolver), or both (pipeguns)! You can also set caps for malfunction chance (so you don't necessarily have to go all the way to 100%), resets on unjamming the gun, and thresholds before certain effects take place!

~~Oh, and I fixed a bug that didn't apply the gun damage multiplier to stamina damage, nyoops.~~

Various changes:

**Detective Revolvers**
Have a baseline misfire chance again. This is to prevent the gun maintenance kit allowing you to avoid misfires altogether. Obviously, just don't make it a .357 if you don't want to roll the dice when you shoot.

**Mosin Nagants/Intern Mosin Nagants**

**Mosins work as they did previous to this change**. I wanted to do some different things, but I'll just keep it as Spark intended for now.

Sawning down the mosin makes them WEIGHT_MEDIUM. Because it's a pistol.

Interns get the Regal Mosin Nagant. It's gold plated and extra moist (and smells of bacon). It also functions as before! Just more noticeably special. And _undulating_.

**Pipeguns/Regal Pipeguns**
Has both jamming and misfiring, but only misfire once they hit the 30% threshold. They also now increment a malfunction chance at 10% instead of 5%. However, their malfunction chance is hardcapped to 30%, and cannot go past this.

Regal Pipeguns cannot jam, as they are intended to be flawless.

They let out a lot of sparks.

**Gun Maintenance Kits**
Now what you use to fix gun malfunction probability as a whole. There is a maint version, which repairs your guns and allows you to reuse the kit (much like the old bundle of cloth would do), and a cargo version, which provides a 5 shot protection from jamming as well as stop the gun gaining malfunction chance for those shots. The premium kits are expended upon use (just like the current gun maint supplies do).

They do NOT prevent misfires. You cannot apply the premium kit to the detective revolver to get around the misfire chance. Just keep them at 25%.

**Modular Receiver Recipe**

It was brought to my attention just how scarce TOOL_SAW actually is, unlike every other tool, because the actual use for TOOL_SAW, which is surgery, has a bunch of unique exceptions in it rather than applying the flag to those items. Kind of silly.

So I kept it to a hatchet for both the receiver and the stock for now. 

## Why It's Good For The Game

Jamming and misfiring were basically doing incredibly similar things, so I decided to do a followup and bring them together into the same relative system, with jamming being a weaker effect to gun malfunctioning, and misfiring being on the extreme end.

## Changelog
:cl:
code: Jamming and misifiring are now utilizing roughly the same system for determining probability! Though not every gun jams or misfires. Also, more specific finetuning for malfunction behaviour.
balance: Gun maintenance kits are now the primary method for gun maintenance. The crafted version reduces malfunction chance back to the base. Premium kits also add a 5 shot grace period from jamming and gun degradation. 
add: Premium kits are expended upon use and do not protect from misfires, only jamming. 
balance: Some rebalancing for probabilities and other stuff for the Mosin Nagant, Obrez, Pipegun and Detective's Revolver.
balance: Modular receivers use hatchets, not saws, for the recipe.
fix: Weapon damage multipliers now also apply to stamina damage. 
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -20. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by necromanceranne** [comment]
@Sparkezel would love your input on this

**Comment by Cyberboss** [comment]
Consider this your last GBP warning

**Comment by Sparkezel** [comment]
I like what you did with the kits so yeah let me adress this pr with bullet points:
>About jamming in mosins:

I dislike that you want to make the jam chance completely disappear after you unjam them, imo it should stay stay as it is as it makes you need to maintain it after a battle, rather than just shoot at a wall so you can unjam it and go back to action.
Mosins jam chance could increase with with bigger but random steps and I would be fine with that
>About harpoon gun:

Its a traitor weapon and it looks recently made so I thought that making jamming affect it was a very bad idea.
>About gun maint kits:

I like what you did in regards to those, personally in regards to jamming/misfire I would rather make them lower the chance to some baseline value (like with mosins it set the misfire chance to 10% instead of 0%) and to affect some sort of multiplier. So if we made mosin `jam_chance` be a random between `(3-7) * quality_modifier` it would set the quality modifier from default 2 to 1 or something in that way

**Comment by Krysonism** [comment]
>Sawning down the mosin gives it a misfire chance

This makes it totally useless.

**Comment by necromanceranne** [comment]
> > About jamming in mosins:
> 
> I dislike that you want to make the jam chance completely disappear after you unjam them, imo it should stay stay as it is as it makes you need to maintain it after a battle, rather than just shoot at a wall so you can unjam it and go back to action.
> Mosins jam chance could increase with with bigger but random steps and I would be fine with that

I actually 100% do not agree with an inconsistent system, so no randomness involved besides the probabilities already at play. I do see your point about unloading into a wall, however. Maybe I should hard cap their jam chance and remove the jam_clear_reverts for now, or only have it halve malfunction chance at most.

> > About gun maint kits:
> 
> I like what you did in regards to those, personally in regards to jamming/misfire I would rather make them lower the chance to some baseline value (like with mosins it set the misfire chance to 10% instead of 0%) and to affect some sort of multiplier. So if we made mosin `jam_chance` be a random between `(3-7) * quality_modifier` it would set the quality modifier from `default 2` to `1` or something in that way

Since it's eating the item to repair the probability chance, I don't see any reason to make this more convoluted than it already is. The guns themselves can carry baseline malfunction chances, so the gun maintenance kit should, at the very least, ensure that it restores them to as they were initially.

> > Sawning down the mosin gives it a misfire chance
> 
> This makes it totally useless. It was already bad...

Just use a gun maintenance kit, it won't jam or increment to the point it misfires.

**Comment by Sparkezel** [comment]
I see what you did in regards to the treshold.I didn't think about any treshold with jamming at all so you can never feel safe unless you cleaned the gun so I would ask you to either lower it a bit or to just remove it unless its still needed somewhere.

**Comment by bluetshirtguy** [comment]
The emergence of rng isn't very fun.

**Comment by necromanceranne** [comment]
> The emergence of rng isn't very fun.

I'm trying to find ways to stop it from being overbearing. At the moment, the moment you start shooting is the moment you start rolling the dice, and that's probably not a good design overall. It should be a gradual, not an immediate, problem.

This is also not every weapon, and only a select few. Not much has actually changed here. It's honestly a pulling back on the severity of these RNG changes so they're not so much of a 'fuck you' out of the blue. You should expect problems once you've used the weapon for a while.

**Comment by necromanceranne** [comment]
I've taken out any changes to the Mosin. They are as they were previously. Including the obrez for now. The only thing done now is that the malfunction system has been folded into the same system as misfiring.

I have also updated the recipe for making modular receivers when a few people told me just how scarce that tool flag actually is, and anything that SHOULD be using that flag (because it's used in the saw surgeries), doesn't. Not exactly my intention there.

**Review by Twaticus — APPROVED** [review]

**Comment by saprasam** [comment]
makes somewhat interesting but shittier version of improvised shotgun
"hmm how could we make it worse?"
"ah yes let's make the gun have a baseline random chance of completely breaking down, preventing you from using it, make another one fucksticks!"
disarms were removed for being essentially random crits
now we have misfire random crits

**Comment by necromanceranne** [comment]
> makes somewhat interesting but shittier version of improvised shotgun
> "hmm how could we make it worse?"
> "ah yes let's make the gun have a baseline random chance of completely breaking down, preventing you from using it, make another one fucksticks!"
> disarms were removed for being essentially random crits
> now we have random fails

I don't know what you mean by 'making them worse' given that currently they're in a much more volatile state because of conflicting mechanics acting on them. All children of the mosin, which includes pipeguns, inherit their jamming stats which are significantly higher than I'd be comfortable with, and largely undetectable by a user. I probably should have just asked Spark to combine our PR's in the first place since we've ended up treading on each other's feet. 

I'm trying to make them use more predictable RNG because right now they're TOO random due to how jamming works. I had a round where my fresh pipegun just jammed every single shot and misfired on a 15%. Then saw zero malfunctions for the rest of the round. Another round I got all the way up to 40% and jammed only once while cleaning out spiders. I didn't intend for the guns to be THAT swingy.

**Comment by ShibaInuLord** [comment]
I feel like the harpoon gun should be in another pr since it makes sense to have it not jam after every shot but don't bunch it together with the other more controversial changes

**Comment by necromanceranne** [comment]
> I feel like the harpoon gun should be in another pr since it makes sense to have it not jam after every shot but don't bunch it together with the other more controversial changes

That's a good point! I'll go ahead and do that.

**Comment by necromanceranne** [comment]
I've readded a method for you to fix your gun from maint. I felt maybe it was a little much to take away any method of fixing your weapon, like you can currently with a sheet of cloth for misfires. Maint gun kits are crafted, are normal sized and don't expend upon use.

The premium kits from cargo work as they do currently, and are expended upon use. They retain the grace period mechanics.

**Comment by AndroidMK2** [comment]
Can you not nerf guns... FOR 5 MINUTES?

**Comment by AndroidMK2** [comment]
Waaah I got killed by a range gun i have to nerf the frick out of them now, me don't like range weapons

**Comment by necromanceranne** [comment]
> Waaah I got killed by a range gun i have to nerf the frick out of them now, me don't like range weapons

You might want to learn how inheritance works before making a comment.

**Comment by necromanceranne** [comment]
Sooo. I had a talk with Rohesie about this and various other changes around these guns. It seems pretty apparent this might have been going entirely in the wrong direction and not in specifically the mechanics itself, but adding mechanics onto items really not worth keeping around as they are.

Their point being that while this is code improvement, it is also potentially sunk cost in that these are weapons that should be removed entirely in the first place. That is, mosins and pipeguns.

Sad as that might be, I actually think the better aim is to nip in the bud the core problem that ballistics are meant to answer (immunity to energy weapons) then sink more code into content that isn't wanted and is meant to being taken out behind the shed.

Basically I got told we want these dead not supported. So uh yeah.
